### PR TITLE
move badger notify after gregor resync

### PR DESCRIPTION
**Pull request by mlsteele**
`Resync` should happen after `serverSync` so that `Resync` gets the latest gregor state.

### CODE REVIEW

**Comment by mmaxim** [comment]
Ok so I think we might need to take a different tack here. I recently that down below the sync in order to fix chat badging, but I see now that it breaks the Gregor based badging. Ideally, we should probably run the chat update where it is now, and then do the Gregor update after that (which probably means running something other than `Resync`.

**Comment by mlsteele** [comment]
Split resync in two. Now it's
1. chat badging
2. Syncer.Connected
3. gregor badging

A side effect is that OnConnected will now send not 1 but 2 badge broadcasts. Can silence the first if that's a problem.

File: go/service/gregor.go [hunk 1]
```diff
@@ -804,6 +809,17 @@ func (g *gregorHandler) OnConnect(ctx context.Context, conn *rpc.Connection,
 		return fmt.Errorf("error running state sync: %s", err)
 	}
 
+	// Update bading from gregor.
```
Comment (mmaxim) [inline go/service/gregor.go]: Typo

File: go/service/gregor.go [hunk 2]
```diff
@@ -804,6 +809,17 @@ func (g *gregorHandler) OnConnect(ctx context.Context, conn *rpc.Connection,
...
+	if g.badger != nil {
+		state, err := gcli.StateMachineState(ctx, nil, false)
+		if err != nil {
+			g.chatLog.Debug(ctx, "unable to get state: %v", err)
```
Comment (mmaxim) [inline go/service/gregor.go]: Can this debug line be a little more specific?

**Pull request merged**